core/src/main/java/io/fabric8/maven/core/service/kubernetes/jib/JibServiceUtil.java

```diff
@@ -128,32 +130,41 @@ public static void jibPush(ImageConfiguration imageConfiguration, MavenProject p
 
         String outputDir = prepareAbsoluteOutputDirPath(EMPTY_STRING, project, outputDirectory).getAbsolutePath();
 
-        ImageName tarImage = new ImageName(imageConfiguration.getName());
-        String tarImageRepo = tarImage.getRepository();
+        String imageName = imageNameFromImageConfiguration(imageConfiguration);
         try {
-            String imageTarName = ImageReference.parse(tarImageRepo).toString().concat(TAR_POSTFIX);
+            String imageTarName = ImageReference.parse(imageName).toString().concat(TAR_POSTFIX);
             TarImage baseImage = TarImage.at(Paths.get(outputDir, imageTarName));
 
             String configuredRegistry = EnvUtil.firstRegistryOf((new ImageName(imageConfiguration.getName())).getRegistry(), imageConfiguration.getRegistry(), registryConfig.getRegistry());
 
             Credential pushCredential = getRegistryCredentials(configuredRegistry, registryConfig);
             final List<String> tags = buildImageConfiguration.getTags();
-            if (tags.isEmpty()) {
-                final String targetImage = new ImageName(imageConfiguration.getName()).getFullName();
-                pushImage(baseImage, targetImage, pushCredential, log);
-            } else {
-                tags.stream().filter(Objects::nonNull).forEach(tag -> {
+
+            appendOriginalImageNameTagIfApplicable(tags, imageName).stream().filter(Objects::nonNull).forEach(tag -> {
                     final String targetImage = new ImageName(imageConfiguration.getName(), tag).getFullName();
                     pushImage(baseImage, targetImage, pushCredential, log);
                 });
-            }
+
         } catch (InvalidImageReferenceException | IllegalStateException e) {
             log.error("Exception occurred while pushing the image: %s", imageConfiguration.getName());
             throw new MojoExecutionException(e.getMessage(), e);
 
         }
     }
 
+    private static List<String> appendOriginalImageNameTagIfApplicable(List<String> tags, String imageName) {
+
+        ImageName tempImage = new ImageName(imageName);
+        Set<String> tagSet = tags.stream().collect(Collectors.toSet());
```

Comment:
Set the filter here instead.
```suggestion
        Set<String> tagSet = tags.stream().filter(Objects::nonNull).collect(Collectors.toSet());
```

Reply:
coolio